### PR TITLE
Remove redundant code

### PR DESCRIPTION
**Pull request by practicalswift**
Remove redundant code.

### CODE REVIEW

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -1016,7 +1016,8 @@ static void json_close(struct command *cmd,
 			return;
 	}
 
-	if (!channel && peer) {
+	if (!channel) {
+		assert(peer);
```
Comment (rustyrussell) [inline lightningd/peer_control.c]: Yech, this points out how confusing this function is.

I think this whole thing needs to be hoist into the `if (peer)` case above.  perhaps as a separate cleanup patch?

**Comment by practicalswift** [comment]
@rustyrussell Agreed, the logic is hard to follow. Will look into `json_close` and submit a separate cleanup patch if I come up with something better.

Removed the `peer_control.c` change from this PR. Please re-review :-)
